reframe/frontend/executors/__init__.py

```diff
@@ -168,15 +175,16 @@ def runall(self, checks, system):
             self._printer.timestamp('Started on', 'short double line')
             self._printer.info()
             self._runall(checks, system)
+            if self._max_retries:
+                self._retry_failed(checks, system)
+
         finally:
-            # Always update statistics and print the summary line
-            self._stats = self._policy.getstats()
-            num_failures = self._stats.num_failures()
+            # Print the summary line
             self._printer.status(
-                'FAILED' if num_failures else 'PASSED',
+                'FAILED' if self._stats.num_failures() else 'PASSED',
                 'Ran %d test case(s) from %d check(s) (%d failure(s))' %
-                (self._stats.num_cases(), len(checks), num_failures),
-                just='center'
+                (self._stats.num_cases(run=0), len(checks),
```

Comment:
You may also move this call out of this statement.

Reply:
Done.